### PR TITLE
feat: get plugin schema by schema_type

### PR DESCRIPTION
**Pull request by liuxiran**
part of https://github.com/apache/apisix/issues/2441

depend on dashboard and apisix modied

- [x] Update plugin's version in package.json

### CODE REVIEW

File: packages/plugin/src/PluginPage.tsx [hunk 1]
```diff
@@ -31,7 +32,7 @@ const PanelSectionStyle = {
 
 const { Sider, Content } = Layout;
 
-const PluginPageApp: React.FC<Props> = ({ initialData = {}, readonly, onChange = () => {} }) => {
+const PluginPageApp: React.FC<Props> = ({ initialData = {}, readonly, onChange = () => {}, schemaType='' }) => {
```
Comment (juzhiyuan) [inline packages/plugin/src/PluginPage.tsx:35]: 可以追加一个默认值

File: packages/plugin/src/PluginPage.tsx [hunk 2]
```diff
@@ -19,6 +19,7 @@ type Props = {
   readonly?: boolean;
   initialData?: PluginPage.FinalData;
   onChange?(data: PluginPage.FinalData): void;
+  schemaType?: string;
```
Comment (juzhiyuan) [inline packages/plugin/src/PluginPage.tsx:22]: 建议补充好类型

File: packages/plugin/src/service.ts [hunk 3]
```diff
@@ -50,10 +50,18 @@ export const getList = async (plugins: Record<string, object> = {}) => {
     });
 };
 
-const cachedPluginSchema: Record<string, object> = {};
-export const fetchPluginSchema = async (name: string): Promise<JSONSchema7> => {
-  if (!cachedPluginSchema[name]) {
-    cachedPluginSchema[name] = (await request(`/schema/plugins/${name}`)).data;
+/** cache pulgin schema by schemaType
```
Comment (juzhiyuan) [inline packages/plugin/src/service.ts:53]: 可以在此使用中文描述

Comment (liuxiran) [reply packages/plugin/src/service.ts:53]: 终于看到中文了，哈哈，英文描述要是没问题的话，还是保持英文吧，风格一致：）
其他问题明天修正，感谢😊

File: packages/plugin/src/PluginPage.tsx (hunk 1 above)
Comment (liuxiran) [reply packages/plugin/src/PluginPage.tsx]: done

File: packages/plugin/src/PluginPage.tsx (hunk 2 above)
Comment (liuxiran) [reply packages/plugin/src/PluginPage.tsx]: done

File: packages/plugin/src/PluginPage.tsx [hunk 4]
```diff
@@ -19,6 +19,7 @@ type Props = {
   readonly?: boolean;
   initialData?: PluginPage.FinalData;
   onChange?(data: PluginPage.FinalData): void;
+  schemaType?: '' | 'route' | 'consumer';
```
Comment (juzhiyuan) [inline packages/plugin/src/PluginPage.tsx:22]: 其实这里保留 `""` 与 `consumer` 就可以啦，当前只有在 consumer 中会特殊处理。如果之后加入了 service，也需要加入插件，那需要再加一个 service 吗hh？

Comment (liuxiran) [reply packages/plugin/src/PluginPage.tsx:22]: 按照 罗泽轩的说法： https://github.com/apache/apisix/issues/2441#issuecomment-710785441
route和service的schema应该是一样的，所以应该不会增加service了，增加route，是为了缓存数据用，我再瞅瞅处理以下 ：）

Comment (juzhiyuan) [reply packages/plugin/src/PluginPage.tsx:22]: 好滴

Comment (juzhiyuan) [reply packages/plugin/src/PluginPage.tsx:22]: 奥看到了，原来还有其它类型的 schema。

**Comment by juzhiyuan** [comment]
Ready to merge?

**Comment by liuxiran** [comment]
> Ready to merge?

depend on: https://github.com/apache/apisix/pull/2509

**Comment by liuxiran** [comment]
> > Ready to merge?
> 
> depend on: [apache/apisix#2509](https://github.com/apache/apisix/pull/2509)

also depend on https://github.com/apache/apisix-dashboard/issues/605

**Comment by juzhiyuan** [comment]
> > > Ready to merge?
> > 
> > 
> > depend on: [apache/apisix#2509](https://github.com/apache/apisix/pull/2509)
> 
> also depend on [apache/apisix-dashboard#605](https://github.com/apache/apisix-dashboard/issues/605)

This is going to resolve in 2.1, any ideas?

**Comment by liuxiran** [comment]
> > > > Ready to merge?
> > > 
> > > 
> > > depend on: [apache/apisix#2509](https://github.com/apache/apisix/pull/2509)
> > 
> > 
> > also depend on [apache/apisix-dashboard#605](https://github.com/apache/apisix-dashboard/issues/605)
> 
> This is going to resolve in 2.1, any ideas?

or let me try to fix it?

**Comment by juzhiyuan** [comment]
> > > > > Ready to merge?
> > > > 
> > > > 
> > > > depend on: [apache/apisix#2509](https://github.com/apache/apisix/pull/2509)
> > > 
> > > 
> > > also depend on [apache/apisix-dashboard#605](https://github.com/apache/apisix-dashboard/issues/605)
> > 
> > 
> > This is going to resolve in 2.1, any ideas?
> 
> or let me try to fix it?

ok, then you go :)

**Comment by juzhiyuan** [comment]
Let me publish this

**Comment by liuxiran** [comment]
> Let me publish this

thx a lot :)

**Comment by juzhiyuan** [comment]
Just published, let me update the dashboard.

**Pull request merged**